subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/internal/DefaultJvmPluginServices.java

```diff
@@ -283,22 +286,19 @@ public static class DefaultElementsConfigurationBuilder extends AbstractConfigur
         private boolean published;
 
         @Inject
-        public DefaultElementsConfigurationBuilder(String name, JvmPluginServices jvmEcosystemUtilities, ConfigurationContainer configurations, SoftwareComponentContainer components, TaskContainer tasks) {
-            super(name, jvmEcosystemUtilities, configurations);
+        public DefaultElementsConfigurationBuilder(String name, JvmPluginServices jvmEcosystemUtilities, ConfigurationContainerInternal configurations, SoftwareComponentContainer components, TaskContainer tasks) {
+            super(name, jvmEcosystemUtilities, (RoleBasedConfigurationContainerInternal) configurations);
             this.components = components;
             this.tasks = tasks;
         }
 
         @Override
         Configuration build() {
-            Configuration cnf = configurations.maybeCreate(name);
+            Configuration cnf = ((RoleBasedConfigurationContainerInternal) configurations).maybeCreateWithRole(name, ConfigurationRoles.INTENDED_CONSUMABLE, false, true);
```

Comment:
Can we do the same for `DefaultResolvableConfigurationBuilder` in this class?

Reply:
Looks like that class might not be used at all and can be completely removed.  Tests will tell.

Reply:
It's deletable.